File: tests/testevo.f90
program main
  implicit none
  call main_()
contains

  subroutine main_()
    use photochem, only: EvoAtmosphere, version, dp
    character(:), allocatable :: err
    type(EvoAtmosphere) :: pc
    integer :: ind
    logical :: converged

    ! Print version
    print*,'photochem version == ',trim(version)

    ! Initialize code
    pc = EvoAtmosphere(&
                      "../data/reaction_mechanisms/zahnle_earth.yaml", &
                      "../examples/ModernEarth/settings.yaml", &
                      "../examples/ModernEarth/Sun_now.txt", &
                      "../examples/ModernEarth/atmosphere.txt", &
                      "../data", &
                      err)
    if (allocated(err)) then
      print*,trim(err)
      stop 1
    endif
    
    converged = pc%find_steady_state(err)
    if (allocated(err)) then
      print*,trim(err)
      stop 1
    endif

  end subroutine

end program

File: tests/testevo.py
from photochem import EvoAtmosphere, zahnle_earth
import numpy as np

from threadpoolctl import threadpool_limits
threadpool_limits(limits=4)

pc = EvoAtmosphere(zahnle_earth,\
                    "testevo_settings.yaml",\
                    "../templates/ModernEarth/Sun_now.txt",\
                    "../templates/ModernEarth/atmosphere_ModernEarth.txt")

t_eval = np.logspace(0,15,10)
success = pc.evolve('test.dat',0.0, pc.var.usol_init, t_eval, True)